### PR TITLE
New package lesstif

### PR DESCRIPTION
**Pull request by DarylGrunau**
This PR provides the lesstif package

### CODE REVIEW

File: var/spack/repos/builtin/packages/lesstif/package.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    depends_on('libxt')
+
+    def setup_build_environment(self, env):
+        env.set('DESTDIR', self.prefix)
+
+    def configure_args(self):
+        args = [
+            '--prefix='
+        ]
+        return args
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/lesstif/package.py]: Can you explain why you needed to set `DESTDIR` and why `--prefix` doesn't work? The [Homebrew package](https://github.com/Homebrew/homebrew-core/blob/master/Formula/lesstif.rb) doesn't do this.

Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/lesstif/package.py]: The --prefix install doesn't get every file into the PREFIX directory so the install fails, e.g.

`5 errors found in build log:
     1290    Making install in autoconf
     1291    make[2]: Entering directory `/tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47
             rbkf3mk/spack-src/scripts/autoconf'
     1292    make[3]: Entering directory `/tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47
             rbkf3mk/spack-src/scripts/autoconf'
     1293    make[3]: Nothing to be done for `install-exec-am'.
     1294    test -z "/usr/share/aclocal" || mkdir -p -- "/usr/share/aclocal"
     1295     /usr/bin/install -c -m 644 'ac_find_motif.m4' '/usr/share/aclocal/ac_find_motif.m4'
  >> 1296    /usr/bin/install: cannot create regular file '/usr/share/aclocal/ac_find_motif.m4': Permission denied
  >> 1297    make[3]: *** [install-aclocalDATA] Error 1
     1298    make[3]: Leaving directory `/tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47r
             bkf3mk/spack-src/scripts/autoconf'
  >> 1299    make[2]: *** [install-am] Error 2
     1300    make[2]: Leaving directory `/tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47r
             bkf3mk/spack-src/scripts/autoconf'
  >> 1301    make[1]: *** [install-recursive] Error 1
     1302    make[1]: Leaving directory `/tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47r
             bkf3mk/spack-src/scripts'
  >> 1303    make: *** [install-recursive] Error 1

See build log for details:
  /tmp/vagrant/spack-stage/spack-stage-lesstif-0.95.2-v2chyxq4oy4mc4sgt22o2cj47rbkf3mk/spack-build-out.txt
`

Comment (DarylGrunau) [reply var/spack/repos/builtin/packages/lesstif/package.py]: Looks like I should incorporate some of the Homebrew build recipe; let me push more changes to this today before merging ...

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/lesstif/package.py]: Yeah, see if installing in serial helps.

**Pull request merged**